include/string.h

```diff
@@ -177,6 +177,12 @@ static inline int strncmp(const char *s1, const char *s2, size_t n) {
     return res;
 }
 
+static inline int string_empty(const char *s) { return !s || !*s; }
+
+static inline int string_equal(const char *s1, const char *s2) {
+    return s1 == s2 || (s1 && s2 && !strcmp(s1, s2));
```

Comment:
Better in what sense? If it's not for a semantic difference, I wouldn't change it.
I'm looking at the compiled code with the latest gcc (see for a simpler version than in the object itself of this repo: https://godbolt.org/z/ETsrY6) and `s1 == s2 || (s1 && s2 && !strcmp(s1, s2));` seems to be generating less code. 
`(!s1 || !s2) ? s1 == s2 : !strcmp(s1, s2);` is executing `cmp     rdi, rsi` twice, as if it doesn't compile away (like in the other case) the `if (s1 == s2)` at the beginning of `strcmp`.
Let me know if I've missed anything here

Reply:
My comment here was merely a nitpick. I was referring to the readability of the code, because construct: s1 && s2 && !strcmp() Strucks me as weird. But that's subjective.

Reply:
Ok, I'll go for your suggestion